src/lib/mina_ledger/lmdb_kvdb.ml

```diff
@@ -0,0 +1,95 @@
+open Core_kernel
+
+module Kvdb = struct
+  type t =
+    { mutable lmdb :
+        ((Bigstring.t, Bigstring.t, [ `Uni ]) Lmdb.Map.t[@sexp.opaque])
+    ; path : string
+    ; uuid : (Uuid.t[@sexp.opaque])
+    }
+  [@@deriving sexp]
+
+  let trivial_conv : Bigstring.t Lmdb.Conv.t =
+    Lmdb.Conv.make ~serialise:(Fn.const Fn.id) ~deserialise:Fn.id ()
+
+  (* TODO is this good enough? what other data can I use to make it more trustably unique? *)
+  let name_to_uuid name =
+    String.to_array name |> Array.map ~f:Char.to_int |> Random.State.make
+    |> Uuid.create_random
+
+  let create (conf : string) =
+    let lmdb =
+      Lmdb.Map.create Nodup (Lmdb.Env.create Rw conf) ~key:trivial_conv
+        ~value:trivial_conv
+    in
+    { lmdb; path = conf; uuid = name_to_uuid conf }
+
+  let close t = Lmdb.Env.close (Lmdb.Map.env t.lmdb)
+
+  let make_checkpoint t name =
+    let env = Lmdb.Map.env t.lmdb in
+    close t ;
+    FileUtil.cp ~recurse:true [ t.path ] name ;
+    t.lmdb <-
+      Lmdb.Map.open_existing Nodup ~key:trivial_conv ~value:trivial_conv env
+
+  let create_checkpoint t name =
+    make_checkpoint t name ;
+    let new_db =
+      Lmdb.Map.open_existing Nodup ~key:trivial_conv ~value:trivial_conv
+        (Lmdb.Env.create Ro name)
+    in
+    { lmdb = new_db
+    ; path = name
+    ; uuid =
+        name_to_uuid name
+        (* Can this be the same, if not what can I use instead of the path? *)
+    }
+
+  let get t ~key =
+    try Some (Lmdb.Map.get t.lmdb key) with Not_found_s _ -> None
+  (* | Not_found -> None *)
+  (* TODO does Not_found_s subsume Not_found ? if not how do I get the compiler to let me do this? *)
```

Comment:
2 comments:
1. [Documentation](https://github.com/Drup/ocaml-lmdb/blob/a392fefea27752dc99698fdf9d85912e3266443b/src/lmdb.mli#L308) seems to state that `get` returns `Not_found` not `Not_found_s`. 
2. The compiler cannot help here. Exceptions are [extensible variants](https://ocaml.org/manual/5.2/extensiblevariants.html#s:extensible-variants), that the compiler cannot check for exhaustiveness. There have been some effort here and there to analyze whether all exceptions were caught (see [this discussion](https://discuss.ocaml.org/t/poor-mans-static-exception-analysis-with-alerts/11296/2))

Reply:
Sorry my comment was not very clear, what I meant was how do I get the compiler not to complain about `Not_found` being deprecated 
```
File "src/lib/mina_ledger/lmdb_kvdb.ml", line 61, characters 9-18:
 > 61 |     with Not_found (* LMDB raises Not_found when the key is absent *) -> ()
 >               ^^^^^^^^^
 > Error (alert deprecated): Not_found
```

Reply:
Understood. [This](https://discuss.ocaml.org/t/section-1-6-exceptions-of-ocaml-manual/12276/2) is the reason. 

The solution here is to disable the alert for this function (and link to the API stating where the `Not_found` comes from).